router/match.go

```diff
@@ -58,10 +60,22 @@ func lookupStringPath(fieldPath []string, obj map[string]interface{}) (string, b
 // is one of `values`. Dots in `field` are interpreted as denoting subobjects
 // -- i.e. the field name "x.y.z" says to check obj["x"]["y"]["z"].
 func fieldMatches(field string, valueMatchers []string, obj map[string]interface{}) bool {
-	val, ok := lookupString(field, obj)
+	val, ok := lookupField(field, obj)
 	if !ok {
 		return false
 	}
+	if val == "" || val == nil {
+		return false
+	}
+	if valueMatchers[0] == "*" {
+		return true
+	}
+	if val == true {
+		val = "true"
+	}
+	if val == false {
+		val = "false"
+	}
```

Comment:
im not confident that this is an acceptable way of doing comparisons with equality. I think you should type assert them before doing any comparisons